coderd/database/queries/provisionerjobs.sql

```diff
@@ -21,8 +21,13 @@ WHERE
 			nested.started_at IS NULL
 			-- Ensure the caller has the correct provisioner.
 			AND nested.provisioner = ANY(@types :: provisioner_type [ ])
-			-- Ensure the caller satisfies all job tags.
-			AND nested.tags <@ @tags :: jsonb
+			-- Ensure the caller satisfies all job tags if requested,
+			AND CASE
+				WHEN @exact_tag_match :: boolean THEN nested.tags = @tags :: jsonb
```

Comment:
Are both guaranteed to be sorted arrays?  Might be a good idea to allow unsorted equality.

Reply:
I think they're `map[string]string` so we want to completely ignore order in comparisons.

Reply:
Ah right, then a `jsonb` to `jsonb` comparison should be fine 👍🏻